<h1>Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: Aaron Faderagao</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing Dataset</p>

<ul><strong>Requirements</strong>
    <li>Name of the Dataframe</li>
    <li>Sources</li>

</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Dataframe</h3>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Selecting a Column</h3>
<h3>Syntax</h3><br>
data_frame['column_name']

In [8]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [9]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [10]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [11]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

<h2>Adelie</h2>

In [12]:
Adelie_df = penguins[penguins['species'] == 'Adelie']

In [13]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [14]:
Gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [15]:
Gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h2>Chinstrap</h2>

In [16]:
Chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [17]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [18]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [19]:
Chinstrap_g = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [20]:
Chinstrap_g.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>Filling Up Null Values</h1>

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [22]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [23]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [24]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [25]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [26]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [27]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [28]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [29]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [30]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [31]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [32]:
penguins['bill_length_mm'], penguins['bill_depth_mm'].mean()

(0      39.10000
 1      39.50000
 2      40.30000
 3      43.92193
 4      36.70000
          ...   
 339    55.80000
 340    43.50000
 341    49.60000
 342    50.80000
 343    50.20000
 Name: bill_length_mm, Length: 344, dtype: float64,
 np.float64(17.151169590643274))

<h1>HOW TO CHANGE THE DATA TYPE OF A COLUMN || AUGUST 7, 2025</h1>

In [33]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [34]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [35]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [36]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>EXPLORATORY DATA ANALYTICS (EDA)</h1>

<h2> QUESTION #1: Which species has longest bill length? </h2>

In [37]:
Chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [38]:
penguins['species'] == 'Adelie'
Adelie_df['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [39]:
penguins['species'] == 'Gentoo'
Gentoo_df['bill_length_mm'].mean()

np.float64(47.50487804878048)

<h4><i>Insights: Based on data the above the Specie of Chinstrap has the longest Bill Length.</h4>

In [40]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
339  4000.000000    male  2009  
340  3400.000000  female  2009  
341  3775.000000    male  2009  
342  4100.000000    male  2009  
343  3775.000000  female  2009  

[344 rows x 8 columns]

<h2> QUESTION #1: Which species of Penguin has shortest flipper length? </h2>

In [41]:
Chinstrap_df['flipper_length_mm'].mean()

np.float64(195.8235294117647)

In [42]:
Adelie_df['flipper_length_mm'].mean()

np.float64(189.95364238410596)

In [43]:
Gentoo_df['flipper_length_mm'].mean()

np.float64(217.1869918699187)

<h4><i>Insights: Based on data the above the Species of Adelie has the shortest flipper length.</h4>
<hr>

<h2> QUESTION #2: Which island had the highest population of penguins? </h2>

In [44]:
penguins["island"].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h4><i>Insights: The island of Biscoe has the highest population of penguins according to the data above and the island with the least population is Torgersen.</h4>
<hr>

<h2> QUESTION #3: How many Male and Female penguins lives in the island of Torgersen? </h2>

In [45]:
penguins[penguins["island"] == "Torgersen"]["sex"].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h4>Insights: According to the data above the penguins that lives in the island of Torgersen have more male penguins than female penguins.</h4>
<hr>

<h2> QUESTION #4: What is the average Bill depth of Chinstrap species? </h2>

In [46]:
penguins[penguins["species"] == "Chinstrap"]["bill_depth_mm"].mean()

np.float64(18.42058823529412)

<h4><i>Insights: Based on data the that we gathered the average bill depth of Chinstrap Species is 18.42</h4>
<hr>

<h2> QUESTION #5: What is the average body mass of Gentoo species?</h2>

In [47]:
avg_gentoo_mass = penguins[penguins["species"] == "Gentoo"]["body_mass_g"].mean()
avg_gentoo_mass

np.float64(5068.965761177136)

<h4><i>Insights: The data shows that the average body mass of Gentoo species is 5068.97 grams.</h3>
<hr>

<h2> QUESTION #6:Which year had the highest number of penguin observations?</h2>

In [48]:
penguins["year"].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h4><i>Insights: According to the data given in this dataset the year 2009 had the most recorded observations unlike year 2007 which had the least pengins observations.</h4>
<hr>

<h2> QUESTION #7: What is the most common penguin species in the dataset?</h2>

In [49]:
most_common_species = penguins["species"].mode()[0]
most_common_species

'Adelie'

<h4><i>Insights: The Adelie species is the most common penguin that can be seen in this dataset compared to other penguin species.</h4>
<hr>

<h2> QUESTION #8: How many penguins lives in each island based on their specie?</h2>

In [50]:
penguin_per_island = penguins.groupby(['species','island']).size()
penguin_per_island

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

<h4><i>Insights: According to the data above there is a significant difference of living penguins based on their specie in each islands and we can see that in each of three islands the species of Adelie is always presented but the Chinstrap species only lives in Dream island while the Gentoo species only lives in Biscoe island.</h4>
<hr>

<h2> QUESTION #9: What is the heaviest average body mass of penguins in each islands based on their specie?</h2>

In [51]:
penguin_avg_mass = penguins.groupby(['species', 'island']) ['body_mass_g'].mean()
penguin_avg_mass

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_g, dtype: float64

<h4><i>Insights: According to the data we can clearly see that the heaviest body mass in Adelie specie is from the Torgersen island. While the Chinstrap specie is on Dream island and the Gentoo specie is on Biscoe island.</h4>
<hr>

<h2> QUESTION #10: Is there a significant difference in observations of penguins in each island per year from 2007-2009?</h2>

In [52]:
penguin_obsrv = penguins.groupby(['island', 'year']).size()
penguin_obsrv

island     year
Biscoe     2007    44
           2008    64
           2009    60
Dream      2007    46
           2008    34
           2009    44
Torgersen  2007    20
           2008    16
           2009    16
dtype: int64

<h4><i>Insights: According to the data there is a significant difference of penguin observations in each island per year. On Biscoe island we can see that the penguins have increased from year 2007 to 2008. And on Dream island, the observation of penguins is high but it suddenly decreased after a year and it increased again on year 2009. While on Torgersen island, there are only 20 penguin observation on year 2007 and on the next two years the number drops to only 16 penguin observations.</h4>
<hr>

<h2> QUESTION #11: Is there a correlation between the bill length and flipper length of penguins?</h2>

In [53]:
penguins_cor1 = penguins["bill_length_mm"].corr(penguins["flipper_length_mm"])
penguins_cor1

np.float64(0.6561813407464275)

<h4><i>Insights: According to the data there is a positive correlation between the bills and flippers of penguins, which would likely indicate that penguins with longer bills also tend to have longer flippers.</h4>
<hr>

<h2> QUESTION #12: Is there a correlation between the penguins bill depth and their body mass??</h2>

In [54]:
penguins_cor2 = penguins["bill_depth_mm"].corr(penguins["body_mass_g"])
penguins_cor2

np.float64(-0.4719156211860662)

<h4><i>Insights: According to the data we can actually see that there is no correlation between the bill depth of penguins to their body mass.</h4>
<hr>

In [55]:
p1 = (penguins[(penguins['species'] == 'Chinstrap') & (penguins['sex'] == 'male')]).info()
p1

<class 'pandas.core.frame.DataFrame'>
Index: 34 entries, 277 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            34 non-null     category
 1   island             34 non-null     category
 2   bill_length_mm     34 non-null     float64 
 3   bill_depth_mm      34 non-null     float64 
 4   flipper_length_mm  34 non-null     float64 
 5   body_mass_g        34 non-null     float64 
 6   sex                34 non-null     category
 7   year               34 non-null     category
dtypes: category(4), float64(4)
memory usage: 2.0 KB
